### ISSUE TITLE
inconsistent SliceTiming conversion of GE epiRT forward/reverse-PE sequences

### ISSUE THREAD

**Issue by hfxcarl** [rordenlab/dcm2niix#723]
**Describe the bug**
Converting GE epiRT sequences with interleaved slice-orders is inconsistent (correct for reverse PE, incorrect for forward PE)

Starting a new fMRI project on a GE 1.5T Signa Artist (vers="30\\LX\\SIGNA_LX1.MR30.0_R01_2242.a") using the epiRT sequence from BrainWaveRT. We are collecting several main task func runs with 197 volumes at TR=1.8sec (pepolar=0), and one rev-PE run with 10 vols at TR=1.8sec (peopler=1). All other sequence parameters of the reverse run matches the forward runs except the pepolar CV. All sequences have the interleaved SliceOrder option enabled in BrainWaveRT. Reviewing individual DICOM slice header RTIA_timer tags (0021,105E) we can verify that in both forward and reverse runs, the slice-order is interleaved as expected. However, when converting with "dcm2niix-v1.0.20220720", only the reverse run outputs the correct interleaved SliceTiming vector in .json file. All forward runs incorrectly output a sequential order in the SliceTiming vector of .json files.

**To reproduce**
1. dcm2niix -b y -x n -z y -o . -f "rev_epi" ./DICOMS/Rev_3/
`{
	"Modality": "MR",
	"MagneticFieldStrength": 1.5,
	"ImagingFrequency": 63.8967,
	"Manufacturer": "GE",
	"InternalPulseSequenceName": "EPI",
	"ManufacturersModelName": "SIGNA Artist",
	"InstitutionName": "-anonymized-",
	"DeviceSerialNumber": "-anonymized-",
	"StationName": "-anonymized-",
	"BodyPartExamined": "BRAIN",
	"PatientPosition": "HFS",
	"ProcedureStepDescription": "-anonymized-",
	"SoftwareVersions": "30\\LX\\SIGNA_LX1.MR30.0_R01_2242.a",
	"MRAcquisitionType": "2D",
	"SeriesDescription": "Main Task TR1800 Rev",
	"ProtocolName": "-anonymized-",
	"ScanningSequence": "EP\\GR",
	"SequenceVariant": "SS",
	"ScanOptions": "EPI_GEMS\\ACC_GEMS",
	"PulseSequenceName": "epiRT",
	"ImageType": ["ORIGINAL", "PRIMARY", "EPI", "NONE"],
	"SeriesNumber": 3,
	"AcquisitionTime": "11:42:17.000000",
	"AcquisitionNumber": 1,
	"SliceThickness": 3.7,
	"SpacingBetweenSlices": 3.7,
	"SAR": 0.0160762,
	"EchoTime": 0.023,
	"RepetitionTime": 1.8,
	"FlipAngle": 90,
	"PhaseEncodingPolarityGE": "Flipped",
	"ShimSetting": [
		0,
		10,
		-276	],
	"PrescanReuseString": "TG/s2,AS/s2,RN/s3,Cal/s3,Coil/s3",
	"CoilString": "19HN",
	"PercentPhaseFOV": 100,
	"PercentSampling": 100,
	"AcquisitionMatrixPE": 64,
	"ReconMatrixPE": 64,
	"ParallelReductionFactorInPlane": 2,
	"EffectiveEchoSpacing": 0.000275556,
	"TotalReadoutTime": 0.01736,
	"PixelBandwidth": 7812.5,
	"PhaseEncodingDirection": "j",
	"SliceTiming": [
		0,
		0.9,
		0.0529412,
		0.952941,
		0.105882,
		1.00588,
		0.158824,
		1.05882,
		0.211765,
		1.11176,
		0.264706,
		1.16471,
		0.317647,
		1.21765,
		0.370588,
		1.27059,
		0.423529,
		1.32353,
		0.476471,
		1.37647,
		0.529412,
		1.42941,
		0.582353,
		1.48235,
		0.635294,
		1.53529,
		0.688235,
		1.58824,
		0.741177,
		1.64118,
		0.794118,
		1.69412,
		0.847059,
		1.74706	],
	"ImageOrientationPatientDICOM": [
		1,
		-0,
		0,
		-0,
		1,
		0	],
	"InPlanePhaseEncodingDirectionDICOM": "COL",
	"ConversionSoftware": "dcm2niix",
	"ConversionSoftwareVersion": "v1.0.20220720"
}`

2. dcm2niix -b y -x n -z y -o . -f "fwd_run-1_epi" ./DICOMS/Func_run1_4/
`{
	"Modality": "MR",
	"MagneticFieldStrength": 1.5,
	"ImagingFrequency": 63.8967,
	"Manufacturer": "GE",
	"InternalPulseSequenceName": "EPI",
	"ManufacturersModelName": "SIGNA Artist",
	"InstitutionName": "-anonymized-",
	"DeviceSerialNumber": "-anonymized-",
	"StationName": "-anonymized-",
	"BodyPartExamined": "BRAIN",
	"PatientPosition": "HFS",
	"ProcedureStepDescription": "-anonymized-",
	"SoftwareVersions": "30\\LX\\SIGNA_LX1.MR30.0_R01_2242.a",
	"MRAcquisitionType": "2D",
	"SeriesDescription": "Main Task TR1800",
	"ProtocolName": "-anonymized-",
	"ScanningSequence": "EP\\GR",
	"SequenceVariant": "SS",
	"ScanOptions": "EPI_GEMS\\ACC_GEMS",
	"PulseSequenceName": "epiRT",
	"ImageType": ["ORIGINAL", "PRIMARY", "EPI", "NONE"],
	"SeriesNumber": 5,
	"AcquisitionTime": "11:44:53.000000",
	"AcquisitionNumber": 1,
	"SliceThickness": 3.7,
	"SpacingBetweenSlices": 3.7,
	"SAR": 0.0160762,
	"EchoTime": 0.023,
	"RepetitionTime": 1.8,
	"FlipAngle": 90,
	"PhaseEncodingPolarityGE": "Unflipped",
	"ShimSetting": [
		0,
		10,
		-276	],
	"PrescanReuseString": "TG/s2,AS/s2,RN/s3,Cal/s3,Coil/s3",
	"CoilString": "19HN",
	"PercentPhaseFOV": 100,
	"PercentSampling": 100,
	"AcquisitionMatrixPE": 64,
	"ReconMatrixPE": 64,
	"ParallelReductionFactorInPlane": 2,
	"EffectiveEchoSpacing": 0.000275556,
	"TotalReadoutTime": 0.01736,
	"PixelBandwidth": 7812.5,
	"PhaseEncodingDirection": "j-",
	"SliceTiming": [
		0,
		0.0529412,
		0.105882,
		0.158824,
		0.211765,
		0.264706,
		0.317647,
		0.370588,
		0.423529,
		0.476471,
		0.529412,
		0.582353,
		0.635294,
		0.688235,
		0.741177,
		0.794118,
		0.847059,
		0.9,
		0.952941,
		1.00588,
		1.05882,
		1.11176,
		1.16471,
		1.21765,
		1.27059,
		1.32353,
		1.37647,
		1.42941,
		1.48235,
		1.53529,
		1.58824,
		1.64118,
		1.69412,
		1.74706	],
	"ImageOrientationPatientDICOM": [
		1,
		-0,
		0,
		-0,
		1,
		0	],
	"InPlanePhaseEncodingDirectionDICOM": "COL",
	"ConversionSoftware": "dcm2niix",
	"ConversionSoftwareVersion": "v1.0.20220720"
}
`

**Expected behavior**
Expecting conversion of all sequences to output correct interleaved slice-timing in .json files.

**Version**
`dcm2niiX version v1.0.20220720  Clang12.0.5 x86-64 (64-bit MacOS)` 

**Troubleshooting**
Tried adding the "--ignore_trigger_times" flag, and it had no effect.
Downloading and using latest released version or compiling dcm2niix has same issue as above.

DropBox Link to DICOMS and output .json files: https://www.dropbox.com/sh/6d8fyn785ou0lc1/AAD_Gc3eC_WHLMjAAxrTOMLKa?dl=0

**Comment by neurolabusc** [comment]
@mr-jaemin do you want to tackle this. I see the same behavior with the [latest stable release (v1.0.20230411)](https://github.com/rordenlab/dcm2niix/releases/tag/v1.0.20230411).

**Comment by mr-jaemin** [comment]
> @mr-jaemin do you want to tackle this. I see the same behavior with the [latest stable release (v1.0.20230411)](https://github.com/rordenlab/dcm2niix/releases/tag/v1.0.20230411).

Sure, will look into this!

**Comment by mr-jaemin** [comment]
Undocumented '-j y' compares GE slice timing from RTIA_timer 0021,105E.  I was able to replicate the issue here:
```
dcm2niix -j y -o . -f fwd Func_run1_4
Chris Rorden's dcm2niiX version v1.0.20230315  GCC9.4.0 x86-64 (64-bit Linux)
undocumented '-j y' compares GE slice timing from 0021,105E
Found 6698 DICOM file(s)
Slice Timing Error between calculated and RTIA timer(0021,105E): 847.118ms
	0	0
	52.9412	900
	105.882	52.9
	158.824	952.9
	211.765	105.8
	264.706	1005.9
	317.647	158.8
	370.588	1058.8
	423.529	211.7
	476.471	1111.7
	529.412	264.7
	582.353	1164.7
	635.294	317.6
	688.235	1217.6
	741.177	370.6
	794.118	1270.6
	847.059	423.5
	900	1323.5
	952.941	476.4
	1005.88	1376.4
	1058.82	529.4
	1111.76	1429.4
	1164.71	582.3
	1217.65	1482.3
	1270.59	635.3
	1323.53	1535.3
	1376.47	688.2
	1429.41	1588.2
	1482.35	741.2
	1535.29	1641.1
	1588.24	794.1
	1641.18	1694.1
	1694.12	847
	1747.06	1747
Convert 6698 DICOM as ./fwd (64x64x34x197)
Conversion required 2.673428 seconds (2.670542 for core code).
```
For epiRT, the slide order (interleaved vs sequential) information comes from DICOM header 
https://github.com/rordenlab/dcm2niix/blob/fe2f26005109f396a4f828aa438241f73fc25fe8/console/nii_dicom_batch.cpp#L6336-L6337 
via  geProtocolBlock 
https://github.com/rordenlab/dcm2niix/blob/fe2f26005109f396a4f828aa438241f73fc25fe8/console/nii_dicom_batch.cpp#L6318


The issue seems be caused by reading `REVSLICEORDER` instead of `SLICEORDER`:

```
$ dicom_protocol_file Func_run1_4/IM-000?-0001.dcm | grep SLICEORDER
REVSLICEORDER "0"
SLICEORDER "1"
$ dicom_protocol_file Rev_3/IM-000?-0001.dcm | grep SLICEORDER
SLICEORDER "1"
```

@neurolabusc Is there a easy way to read the exact string `SLICEORDER` instead of substring?

**Comment by neurolabusc** [comment]
@mr-jaemin well spotted. How about adding an end of line character at the start of the [SLICEORDER key](https://github.com/rordenlab/dcm2niix/blob/fe2f26005109f396a4f828aa438241f73fc25fe8/console/nii_dicom_batch.cpp#L997) so it reads:

```
char keyStrSO[] = "\nSLICEORDER";
```

**Comment by mr-jaemin** [comment]
> @mr-jaemin well spotted. How about adding an end of line character at the start of the [SLICEORDER key](https://github.com/rordenlab/dcm2niix/blob/fe2f26005109f396a4f828aa438241f73fc25fe8/console/nii_dicom_batch.cpp#L997) so it reads:
> 
> ```
> char keyStrSO[] = "\nSLICEORDER";
> ```

Great idea!

**Comment by neurolabusc** [comment]
@hfxcarl can you close this issue if the latest commit to the development branch resolves your issue (v1.0.20230623):

```
git clone --branch development https://github.com/rordenlab/dcm2niix.git
cd dcm2niix/console
make
./dcm2niix
```

**Comment by hfxcarl** [comment]
@neurolabusc and @mr-jaemin Yes, this latest commit does indeed resolve the issue. Thank you both very much!
